File: workbook-6/demos/01-interfaces-intro-final/src/main/java/com/pluralsight/models/Vehicle.java
package com.pluralsight.models;

import com.pluralsight.models.interfaces.Drivable;

public abstract class Vehicle implements Drivable
{
    private String make;
    private String model;
    private boolean isOn;

    public Vehicle(String make, String model)
    {
        this.make = make;
        this.model = model;
        this.isOn = false;
    }

    @Override
    public String getMake()
    {
        return make;
    }

    @Override
    public String getModel()
    {
        return model;
    }


    @Override
    public void turnOn()
    {
        this.isOn = true;
        System.out.println("Turning on Vehicle");
    }

    @Override
    public void turnOff()
    {
        this.isOn = false;
        System.out.println("Turning off Vehicle");
    }
}

File: workbook-6/demos/02-review-python/vehicle.py
class Vehicle:
    # no such thing as private
    make = ""
    model = ""
    isOn = False
    
    # constructor
    def __init__(self, make, model):
        self.make = make
        self.model = model
        print(f"building a new {make} {model} - (Vehicle constructor)")

    # methods
    def turnOn(self):
        self.isOn = True
        
    def turnOff(self):
        self.isOn = False
    
    def repair(self):
        print(f"repairing {self.make} {self.model} (code inside Vehicle super class)")
        